xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -349,7 +390,35 @@ document.observe('xwiki:dom:loaded', function() {
   if ($('documentsTrash')) {
     $('documentsTrash').up('.hidden').removeClassName('hidden');
   }
-});</code>
+});
+
+// we need to wait the complete display of the livetable to get the number of rows
+document.observe('xwiki:livetable:documentsTrash:displayComplete', function() {
+  if ($('documentsTrash')) {
+    var actionAllContainer = $('documentsTrash').
+        getElementsBySelector('td.xwiki-livetable-display-header-filter:last-child')[0];
+
+    if (trashDocumentsLivetable.totalRows > 0) {
+      var trashAllClass = 'btn btn-danger';
+    } else {
+      var trashAllClass = 'btn btn-danger disabled';
+    }
+
+    // avoid to get multiple times the button
+    if ($('trashAllDeleted')) {
+      $('trashAllDeleted').writeAttribute('class', trashAllClass);
+    } else {
+      var trashAll = new Element('a', {
+          'id': 'trashAllDeleted',
+          'class': trashAllClass,
+          'href':'#'
+          }).update('$services.localization.render('index.deletedDocuments.deleteAll')');
```

Comment:
You don't get any warning in the server log that the JSX can't be minified? Might be because the JSX is evaluated (Velocity) first and then minified, but that's not the case with JavaScript code coming from the WAR. It's always good to ensure the JavaScript remains valid when you mix it with Velocity.

Reply:
Hm my bad I should have used double quotes. But I did not see any warning.